lib/api/objectPut.js

```diff
@@ -46,6 +46,10 @@ function _storeIt(bucketName, objectKey, objMD, authInfo, canonicalID,
         dataToDelete = Array.isArray(objMD.location) ?
             objMD.location : [objMD.location];
     }
+    // not deleting old data in the case of versioning
+    if (objMD && objMD['x-amz-version-id'] !== 'null') {
+        dataToDelete = undefined;
+    }
```

Comment:
I see now that objMD['x-amz-version-id'] was set to 'null' in current implementation so ok on backwards compatibility but shouldn't we still be able to delete data if versioning is not enabled?